### ISSUE TITLE
scipy.ndimage.convolve doesn't work when output parameter is set to input

### ISSUE THREAD

**Issue by skiguy0123** [scipy/scipy#3694]
In place convolution fails

```
import numpy as np
from scipy.ndimage import convolve
ker = [1., 2., 1.]
x = np.random.random_sample(100)
out1 = convolve(x, ker)
out2 = convolve(x, ker)
convolve(x, ker, output=x)
print np.allclose(out1, out2)
>>> True
print np.allclose(out1, x)
>>> False
```

**Comment by rgommers** [comment]
Makes sense to add a sentence to the ``output`` parameter description for each function in ``ndimage`` that has this keyword.

**Comment by adivijaykumar** [comment]
Hi, would like to work on this issue.

So is it sure that this problem arises due to the aliasing? Shall I proceed with what @rgommers has suggested?

**Comment by rgommers** [comment]
> So is it sure that this problem arises due to the aliasing? 

Looks like it.

> Shall I proceed with what @rgommers has suggested?

A PR for that would be nice. Try to keep it as short as possible while still being understandable.

**Comment by adivijaykumar** [comment]
This should be closed, shouldn't it?

**Comment by ev-br** [comment]
Closing then. (Next time, add a magic line 'fixes gh-XXXX` to the PR or a commit message)

**Comment by adivijaykumar** [comment]
Okay, will remember this next time. Thanks!

**Comment by kalvdans** [comment]
The documentation changes were reverted in https://github.com/scipy/scipy/pull/8528, so let's reopen this issue, shall we?

**Comment by j-bowhay** [comment]
@kalvdans would you be interested in submitting a pr, it looks like it would simply be a case of transferring part of the old doc

**Comment by kalvdans** [comment]
> @kalvdans would you be interested in submitting a pr, it looks like it would simply be a case of transferring part of the old doc

I can try, but I think we can do better, by adding a runtime check such as `assert not np.may_share_memory(input, output)`. What do you think, dear issue subscribers?

**Comment by JohanSchott** [comment]
I think https://github.com/scipy/scipy/pull/11986 fixed this issue, by using a temporary buffer if `output=input`. 
So close this issue?
